City-of-Helsinki/drupal-hdbt
# UHF-8136 accordion accessibility

# [UHF-8136](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8136)

## What was done

- Changes based on [accessibility report](https://helsinkisolutionoffice.atlassian.net/wiki/spaces/HEL/pages/8212119607/22.2.2023+Ty+lista#%E2%9C%85-Haitarit)
- Fixed bug causing error when loading a page with non existing anchor link

## How to install

* Make sure your instance is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-8136_accordion_accessibility`
* Run `make drush-cr`

## How to test
* Go to a page with accordion
  * Add gibberish as anchor link (#qwerty) and reload the page
  * Check inspector for javascript errors. No errors should emerge from accordion library.
* Test accessibility:
  * Hidden span (with "toggle accordion" -text for screen readers) shuld removed from accordion item header
  * Aria-labelled-by removed from toggle button
  * Aria-expanded removed from close button
  * Close button's "keypress" or "mouseup" event should transfer the focus back to the accordion item's toggle button
 

## Designers review



## Other PRs


* Link to other PR
